common/components/controllers/EditProjectController.jsx

```diff
@@ -0,0 +1,39 @@
+// @flow
+
+import React from 'react';
+import EditProjectForm from '../common/projects/EditProjectForm.jsx'
+
+type State = {|
+  projectId: number
+|};
+
+/**
+ * Encapsulates form for editing projects
+ */
+class EditProjectController extends React.PureComponent<{||},State> {
+  constructor(props: Props): void {
+    super(props);
+  
+    this.state = {
+      projectId:  (new RegExp("id=([^&]+)")).exec(document.location.search)[1]
```

Comment:
this seems kinda funky, like it should be gotten from a Prop or a Store or at least a Util devoted to URL inspection

Reply:
Moved to url library